test/test-addon.js

```diff
@@ -52,6 +64,23 @@ test('build simple addon', function (t) {
   proc.stderr.setEncoding('utf-8')
 })
 
+test('make sure addon symbols do not overlap', function (t) {
+  t.plan(3)
+
+  var addonPath = path.resolve(__dirname, 'node_modules', 'duplicate_symbols')
+  // Set the loglevel otherwise the output disappears when run via 'npm test'
+  var cmd = [nodeGyp, 'rebuild', '-C', addonPath, '--loglevel=verbose']
+  var proc = execFile(process.execPath, cmd, function (err, stdout, stderr) {
+    var logLines = stderr.toString().trim().split(/\r?\n/)
+    var lastLine = logLines[logLines.length-1]
+    t.strictEqual(err, null)
+    t.strictEqual(lastLine, 'gyp info ok', 'should end in ok')
+    t.strictEqual(runDuplicateBindings().trim(), 'not equal')
+  })
+  proc.stdout.setEncoding('utf-8')
+  proc.stderr.setEncoding('utf-8')
```

Comment:
The setEncoding() calls are no-ops since `proc.stdout` and `proc.stderr` aren't otherwise used.